lib/3scale_toolbox/tasks/update_service_settings_task.rb

```diff
@@ -11,14 +11,30 @@ def initialize(source:, target:, target_name:)
 
       def call
         source_obj = source.show_service
-        response = target.update_service(target_service_params(source_obj))
-        raise ThreeScaleToolbox::Error, "Service has not been saved. Errors: #{response['errors']}" unless response['errors'].nil?
+        svc_obj = update_service target_service_params(source_obj)
+        if (errors = svc_obj['errors'])
+          raise ThreeScaleToolbox::Error, "Service has not been saved. Errors: #{errors}" \
+        end
 
         puts "updated service settings for service id #{source.id}..."
       end
 
       private
 
+      def update_service(service)
+        svc_obj = target.update_service service
+
+        # Invalid deployment option check
+        # use default deployment_option
+        if (errors = svc_obj['errors']) &&
+           ThreeScaleToolbox::Helper.service_invalid_deployment_option?(errors)
+          service.delete('deployment_option')
```

Comment:
I think might be good to clarify why the deployment_option is deleted and then the update retried.
It's hard to know why is being done for that parameter only and not for others

Reply:
:+1: